### PR TITLE
Reduce output when resizing to avoid breaking the progress

### PR DESCRIPTION
**Pull request by nohwnd**
Before: when the window gets smaller the progress is broken, and end up scrolling down on the screen. Same thing happens when assembly name is too long.

sizing-broken

After: we hide the tfm and architecture first, then we start trimming the name from right prepending with ... to keep the output to correct width and keep the time on screen.

sizing

### CODE REVIEW

**Review by MarcoRossignoli — APPROVED** [review]

File: test/UnitTests/Microsoft.Testing.Extensions.UnitTests/CrashDumpTests.cs [hunk 1]
```diff
@@ -19,6 +19,7 @@ public class CrashDumpTests(ITestExecutionContext testExecutionContext) : TestBa
     [Arguments("Full")]
     public async Task IsValid_If_CrashDumpType_Has_CorrectValue(string crashDumpType)
     {
+        Thread.Sleep(20_000);
```
Comment (Evangelink) [inline test/UnitTests/Microsoft.Testing.Extensions.UnitTests/CrashDumpTests.cs]: Should be reverted

**Comment by Evangelink** [comment]
What about having the time just after the tfm/platform? If I have a really long line there might not be much need to have so much empty space and it seems to be causing more complexity that could be avoided.

File: test/UnitTests/Microsoft.Testing.Extensions.UnitTests/CrashDumpTests.cs (hunk 1 above)
Comment (nohwnd) [reply test/UnitTests/Microsoft.Testing.Extensions.UnitTests/CrashDumpTests.cs]:
```suggestion
```

**Comment by nohwnd** [comment]
It looks worse, and it does not reduce complexity. 

It would be as complex. 

Now we write the duration at the end of the line by jumping to x chars from the end of the line and writing the string. When we see that the progress state was not updated (there is .Updated property on it that counts up), we just write over the previous duration without writing the whole line again. If we put the duration after the assembly name we need to remember on which offset it is (same amount of work as before, just different value).

And we won't save the complexity of calculating how long that duration string is, because we need to "reserve" space for it if we want to always see it on the screen.

**Comment by nohwnd** [comment]
Do you see any other complexity that could be avoided?

**Comment by nohwnd** [comment]
There is also max console width, if you have super wide window we don't go past 250 chars or something.

**Review by Evangelink — APPROVED** [review]

**Pull request merged**